File: src/main/scala/fish/philwants/modules/LinkedinModule.scala
package fish.philwants.modules

import fish.philwants.Credentials
import fish.philwants.JsoupImplicits._


object LinkedinModule extends AbstractModule {
  override val moduleName: String = "LinkedIn"
  override val uri: String = "https://www.linkedin.com"

  // Given credentials return a LoginResult
  override def tryLogin(creds: Credentials): Boolean = {
    val resp = get(uri).execute()

    // Get the form and update it with credentials
    val form = resp.selectForm("form.login-form")
      .update("session_key", creds.username)
      .update("session_password", creds.password)

    // Try to login
    val loginResp = form
      .submit()
      .cookies(resp.cookies())
      .followRedirects(false)
      .execute()

    // Check login result
    loginResp.statusCode() == 302
  }
}

File: secondary_implementations/python/modules/LinkedinModule.py
# vim: ts=2 sw=2 noexpandtab
import requests
from lxml import html
from .AbstractModule import AbstractModule

class LinkedinModule(AbstractModule):
	def try_login(self, uname, password):
		headers = {'user-agent': AbstractModule().user_agent()}
		resp = requests.get('https://www.linkedin.com', headers=headers)

		tree = html.fromstring(resp.content)
		payload = tree.forms[0].fields
		payload = {k : v for k, v in payload.items()}
		payload['session_key'] = uname
		payload['session_password'] = password
		loginResp = requests.post(
			'https://www.linkedin.com/uas/login-submit', data=payload,
			headers=headers, cookies=resp.cookies, allow_redirects=False)

		return loginResp.status_code == 302

linkedin = LinkedinModule()
